src/services/bot/utils/getNotifications.ts

```diff
@@ -27,6 +28,15 @@ export async function* getNotifications(
       const addressJettonsFromDb = await getJettonsFromDB(wallet.id)
       for (const jetton of addressJettonsFromDb) {
         if (!addressJettonsFromChainObj[jetton.token]) {
+          const firstPurchase = await getFirstAddressJettonPurchaseFromDB(jetton.id)
+          if (!firstPurchase) {
```

Comment:
Is this possible? If we think it's not, we need to log this.

Reply:
This is technically possible. if we have a race condition.
I replaced multiple queries with a single postgres transaction with these queries to completly avoid that possibility now.